scrapper.py

```diff
@@ -21,77 +39,245 @@ class IncorrectNumberOfArticlesError(Exception):
     """
 
 
-class UnknownConfigError(Exception):
-    """
-    Most general error
+class IncorrectStatusCode(Exception):
+    """"
+    Custom error
     """
 
 
+headers = {
+        'user-agent':
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 '
+            '(KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36'}
+
+
 class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls: list, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls: list, max_articles: int, max_articles_per_seed: int):
+        self.seed_urls = seed_urls
+        self.max_articles_per_seed = max_articles_per_seed
+        self.max_articles = max_articles
+        self.urls = []
 
     @staticmethod
     def _extract_url(article_bs):
-        pass
+        page_links = []
+        for tag_a_content in article_bs.find_all('a'):
+            get_url = tag_a_content.get('href')
+            if get_url:
```

Comment:
to nested. simplify a bit by using continue and break statements. your code should not be so deep